src/nwb/ecephys/SpikeEventSeries.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <string>
+
+#include "BaseIO.hpp"
+#include "Channel.hpp"
+#include "nwb/ecephys/ElectricalSeries.hpp"
+
+namespace AQNWB::NWB
+{
+/**
+ * @brief Stores snapshots/snippets of recorded spike events (i.e., threshold
+ * crossings).
+ */
+class SpikeEventSeries : public ElectricalSeries
+{
+public:
+  /**
+   * @brief Constructor.
+   * @param path The location of the SpikeEventSeries in the file.
+   * @param io A shared pointer to the IO object.
+   * @param description The description of the SpikeEventSeries, should describe
+   * how events were detected.
+   */
+  SpikeEventSeries(const std::string& path,
+                   std::shared_ptr<BaseIO> io,
+                   const BaseDataType& dataType,
+                   const Types::ChannelVector& channelVector,
+                   const std::string& description,
+                   const SizeArray& dsetSize,
+                   const SizeArray& chunkSize,
+                   const float& conversion = 1.0f,
+                   const float& resolution = -1.0f,
+                   const float& offset = 0.0f);
+
+  /**
+   * @brief Destructor
+   */
+  ~SpikeEventSeries();
+
+  /**
+   * @brief Initializes the Electrical Series
+   */
+  void initialize();
+
+  /**
+   * @brief Write a single spike series event
+   * @param numSamples The number of samples in the event
+   * @param numChannels The number of channels in the event
+   * @param data The data of the event
+   * @param timestamps The timestamps of the event
+   * @param
+   */
+  Status writeSpike(const SizeType& numSamples,
+                    const SizeType& numChannels,
+                    const void* data,
+                    const void* timestamps);
+
+private:
+  /**
+   * @brief The neurodataType of the SpikeEventSeries.
+   */
+  std::string neurodataType = "SpikeEventSeries";
```

Comment:
Note, with #91 we won't need to store the neurodataType, because each of these classes is being registered and will automatically get a function to get the name and namespace for the type.

Reply:
Sounds great!